src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -289,6 +289,39 @@
  * }
  * </pre>
  * <p>
+ * To configure the check to target fields types only:
+ * </p>
+ * <pre>
+ * &lt;module name="IllegalType"&gt;
+ *   &lt;property name=&quot;illegalClassNames&quot; value=&quot;java.util.Optional&quot;/&gt;
+ *   &lt;property name=&quot;tokens&quot; value=&quot;VARIABLE_DEF&quot;/&gt;
+ *   &lt;property name=&quot;id&quot; value=&quot;IllegalTypeOptionalAsField&quot;/&gt;
+ * &lt;/module&gt;
+ * &lt;module name="SuppressionXpathSingleFilter"&gt;
+ *   &lt;property name=&quot;query&quot; value=&quot;//METHOD_DEF//*&quot;/&gt;
+ *   &lt;property name=&quot;id&quot; value=&quot;IllegalTypeOptionalAsField&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <pre>
+ * import java.util.Optional;
+ *
+ * public class Main {
+ *
+ *   private Optional&lt;String&gt; field1; // violation, usage of type 'Optional' is not allowed
+ *   static int field2 = 4; // OK
+ *   protected String field3; // OK
+ *   public Optional&lt;String&gt; field4; // violation, usage of type 'Optional' is not allowed
+ *   Optional&lt;String&gt; field5; // violation, usage of type 'Optional' is not allowed
```

Comment:
Take a look at https://checkstyle.sourceforge.io/config_coding.html#DeclarationOrder.  It would make this easier to look at if these field declarations followed this style.  Please change the `field*` number accordingly.

Reply:
```
static int field1 = 4; // OK
  public Optional<String> field2; // violation, usage of type 'Optional' is not allowed
  protected String field3; // OK
  Optional<String> field4; // violation, usage of type 'Optional' is not allowed
  private Optional<String> field5; // violation, usage of type 'Optional' is not allowed
```
is this correct order?

Reply:
This can be verified by using the check.

Reply:
done

Reply:
Resolved.